checks.d/mysql.py

```diff
@@ -141,134 +387,241 @@ def _connect(self, host, port, mysql_sock, user, password, defaults_file):
-
-            self.gauge("mysql.innodb.buffer_pool_free", innodb_buffer_pool_pages_free, tags=tags)
-            self.gauge("mysql.innodb.buffer_pool_used", innodb_buffer_pool_pages_used, tags=tags)
-            self.gauge("mysql.innodb.buffer_pool_total", innodb_buffer_pool_pages_total, tags=tags)
-
-            if innodb_buffer_pool_pages_total != 0:
-                innodb_buffer_pool_pages_utilization = \
-                    innodb_buffer_pool_pages_used / innodb_buffer_pool_pages_total
-                self.gauge("mysql.innodb.buffer_pool_utilization",
-                           innodb_buffer_pool_pages_utilization, tags=tags)
-
-        if 'galera_cluster' in options and options['galera_cluster']:
-            value = self._collect_scalar('wsrep_cluster_size', status_results)
-            self.gauge('mysql.galera.wsrep_cluster_size', value, tags=tags)
-
-        if 'replication' in options and options['replication']:
+        key_blocks_unused = self._collect_scalar('Key_blocks_unused', results)
+        key_cache_block_size = self._collect_scalar(
+            'key_cache_block_size', results)
+        key_buffer_size = self._collect_scalar('key_buffer_size', results)
+        key_cache_utilization = 1 - \
+            ((key_blocks_unused * key_cache_block_size) / key_buffer_size)
+
+        results['Key_buffer_size'] = key_buffer_size
+        results['Key_buffer_bytes_used'] = self._collect_scalar(
+            'Key_blocks_used', results) * key_cache_block_size
+        results['Key_buffer_bytes_unflushed'] = self._collect_scalar(
+            'Key_blocks_not_flushed', results) * key_cache_block_size
+        results['Key_cache_utilization'] = key_cache_utilization
+
+        metrics.update(VARIABLES_VARS)
+        metrics.update(INNODB_VARS)
+        metrics.update(BINLOG_VARS)
+
+        if options.get('extra_status_metrics'):
+            self.log.debug("Collecting Extra Status Metrics")
+            metrics.update(OPTIONAL_STATUS_VARS)
+
+            if self._version_compatible(db, host, "5.6.6"):
+                metrics.update(OPTIONAL_STATUS_VARS_5_6_6)
+
+        if options.get('galera_cluster'):
+            # already in result-set after 'SHOW STATUS' just add vars to collect
+            self.log.debug("Collecting Galera Metrics.")
+            metrics.update(GALERA_VARS)
+
+        if options.get('extra_performance_metrics') and self._version_compatible(db, host, "5.6.0"):
+            # report avg query response time per schema to Datadog
+            try:
+                results['perf_digest_95th_percentile_avg_us'] = self._get_query_exec_time_95th_us(db)
+                results['query_run_time_avg'] = self._query_exec_time_per_schema(db)
+                metrics.update(PERFORMANCE_VARS)
+            except Exception as e:
+                self.log.debug("Performance metrics unavailable at this time: {0}".format(e))
+
+        if options.get('schema_size_metrics'):
```

Comment:
I would suggest highlighting in the user documentation of the check, the possibility of performance problems in case a user has large number of schemas and/or tables.